src/pages/single-product/single-product.tsx

```diff
@@ -0,0 +1,95 @@
+import { useContext, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { FALLBACK_IMAGE, PRODUCTS_ENDPOINT } from "../../constants";
+import { CartContextObject } from "../../entities/cart-context-object";
+import useProduct from "../../hooks/use-product";
+import { CartContext } from "../../providers/cart-provider";
+import { ProductContext } from "../../providers/products-provider";
+import APIClient from "../../services/api-client";
+import CartService from "../../services/cart-service";
+
+const SingleProduct = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  const { data: product, error, isLoading } = useProduct(id!);
```

Comment:
I honestly don't like the non-null assertion (id!) and I would rather do a check before.
Check this out: https://blog.italanta.net/the-case-against-non-null-assertion-operator-ea6848aed636